### ISSUE TITLE
Using a modern dev setup (webpack, Babel and so on)

### ISSUE THREAD

**Issue by cdelaorden** [clientIO/joint#334]
Hi!

Has anyone succeeded in trying to use a modern dev stack for development? The idea of having N scripts loaded via SCRIPT tag and using the global namespace is more or less deprecated today. I'd like to have a single JS bundle with all dependencies included.

I have managed to setup JointJS with webpack, Babel, but trying to add the Rappid add-on (evaluation version) to the mix is impossible, as it tries to find "joint" in the global namespace,  then fails because it can't find module `ws` for example (for Channel I guess).

I suppose no one is doing a medium/big project with one SCRIPT tag per file, so what are the solutions? Require.js. which is mentioned in the documentation, has become a thing  of the past as well. 

Any hints, or examples, or is this completely unsupported?

**Comment by chill117** [comment]
We have a separate repository for the Rappid plugins, but it is structured similarly to this one (JointJS). In our unit tests for both Rappid and JointJS, we include some sanity checks for webpack and browserify bundles. You can see how we build the webpack bundle [here](https://github.com/clientIO/joint/blob/master/Gruntfile.js#L110-L128). We are using [grunt-webpack](https://github.com/webpack/grunt-webpack) with webpack version 1.13.1. The end result is exactly what you are describing - a totally self contained version of Rappid or JointJS that only exposes `joint` on the global namespace.

**Comment by quentin-sommer** [comment]
Hi, I'm trying as well to make jointjs work with webpack, my problem I get the error
```Module not found: Error:
Can't resolve 'fs' in '/proj/node_modules/jointjs' @ ./~/jointjs/index.js 3:9-22```

I guess because jointjs tries to load node's fs module:
``` javascript
'use strict';

var fs = require('fs');

var possiblePaths = ['./build/joint.js', './dist/joint.js'];
var filePath;

while ((filePath = possiblePaths.shift())) {
    try {
        fs.statSync(__dirname + '/' + filePath);
    } catch (error) {
        // Try another path.
        continue;
    }
    // Found a path that exists.
    break;
}
console.log('hey',filePath);
if (!filePath) {
    throw new Error('JointJS build file not found.');
}
module.exports = require(filePath);
```
I have tried these workarounds:
``` javascript
node: {
    fs: 'empty'
  },
```
``` javascript
node: {
    fs: true
  },
```
But neither worked as the first mocks the module (and then make the code fail) and the other make webpack output `Error: No browser version for node.js core module 'fs' available`

How can I work around this to use jointjs ? @chill117

**Comment by chill117** [comment]
This has been fixed in [v0.9.10](https://github.com/clientIO/joint/releases/tag/v0.9.10)
